src/mon/ConfigMonitor.cc

```diff
@@ -647,22 +647,23 @@ void ConfigMonitor::load_config()
     }
 
     string section_name;
-    if (key.find("mgr") == 0) {
-      name = key.substr(key.find('/') + 1);    
-      MaskedOption mopt(new Option(name, Option::TYPE_STR, Option::LEVEL_UNKNOWN));
+    if (key.find("/mgr/") != std::string::npos) {
+      // mgr module option, "something/mgr/foo"
+      name = key.substr(key.find("/mgr/") + 1);
+      MaskedOption mopt(new Option(name, Option::TYPE_STR,
+				   Option::LEVEL_UNKNOWN));
       mopt.raw_value = value;      
       Section *section = &config_map.global;;
       section_name = "mgr";
-      if (section_name.size()) {
-	if (section_name.find('.') != std::string::npos) {
-	  section = &config_map.by_id[section_name];
-	} else {
-	  section = &config_map.by_type[section_name];
-	}
+      if (section_name.find('.') != std::string::npos) {
```

Comment:
`section_name` is always `"mgr"` here, shall we get the section name using `ConfigMap::parse_mask()` just like how we handle the normal options?

Reply:
This also bugged me when I was reading the code.  These options are meaningless outside mgr, but for consistency setting a mgr option on the osds should have no effect.  I'll add a patch to fix this.